File: src/corelibs/seguridad.py
"""Operaciones sencillas de seguridad y hashing."""

import hashlib
import uuid
import warnings


def hash_sha256(texto: str) -> str:
    """Devuelve el hash SHA-256 de *texto*."""
    return hashlib.sha256(texto.encode("utf-8")).hexdigest()


# MD5 es inseguro y se mantiene sólo como alias de ``hash_sha256`` para
# compatibilidad con versiones anteriores.
hash_md5 = hash_sha256


def generar_uuid() -> str:
    """Genera un identificador único."""
    return str(uuid.uuid4())

File: src/pcobra/core/nativos/seguridad.js
import crypto from 'crypto';
import { v4 as uuidv4 } from 'uuid';

export function hash_sha256(texto) {
    return crypto.createHash('sha256').update(texto).digest('hex');
}

// ``hash_md5`` se mantiene como alias a ``hash_sha256`` por compatibilidad.
export const hash_md5 = hash_sha256;

export function generar_uuid() {
    return uuidv4();
}
